packages/@react-stately/virtualizer/src/Virtualizer.ts

```diff
@@ -877,9 +877,11 @@ export class Virtualizer<T extends object, V, W> {
     if (this._transaction) {
       for (let view of this._transaction.toRemove.values()) {
         let cur = view.layoutInfo;
-        let layoutInfo = this.layout.getLayoutInfo(cur.key);
-        if (this._applyLayoutInfo(view, layoutInfo)) {
-          updated = true;
+        if (cur?.key !== undefined) {
```

Comment:
we want to catch both null and undefined
```
null !== undefined
true
null != undefined
false
```
it's preferable when checking keys to use `cur?.key != null` as a result